### PR TITLE
Add WebDAV to API2:

### PR DESCRIPTION
**Pull request by jdrossl**
- Add new controller to expose the existing service
- Small refactor for request constants used by other services

https://github.com/craftercms/craftercms/issues/3420

### CODE REVIEW

File: src/main/api/2/studio-api-2.yaml [hunk 1]
```diff
@@ -1325,6 +1327,105 @@ paths:
...
+          schema:
+            type: integer
+            format: int64
+        - name: path
+          in: query
+          description: The path of the directory to list (defaults to the root)
+          required: false
+          schema:
+            type: string
+        - name: type
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml:1358]: Either switch to MIME-type or spec out the types.

File: src/main/api/2/studio-api-2.yaml [hunk 2]
```diff
@@ -1325,6 +1327,105 @@ paths:
...
+              schema:
+                type: object
+                properties:
+                  response:
+                    $ref: '#/components/schemas/ApiResponse'
+                  items:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/WebDavItem'
+        '401':
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml:1378]: Add 404

Comment (jdrossl) [reply src/main/api/2/studio-api-2.yaml:1378]: Actually the WebDAV client never fails, it just returns an empty list

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Please add permissions per our convo.

**Review by avasquez614 — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**